modules/management-group-bootstrap/readers.tf

```diff
@@ -15,10 +15,26 @@ resource "azurerm_role_assignment" "readers" {
   role_definition_name = "Reader"
 }
 
-#resource "azurerm_role_assignment" "app_gateway_backend_health_reader" {
-#for_each = var.groups
+locals {
+  custom_role_assignments = [
+    "Application Gateway Backend Health Reader"
+  ]
 
-#principal_id         = azuread_group.readers[each.value.id].object_id
-#scope                = "/providers/Microsoft.Management/managementGroups/${each.value.id}"
-#role_definition_id   = "159469f4-6e46-4b09-aa73-8e2f243aa784"
-#}
+  role_assignments = distinct(flatten([
+    for group in var.groups : [
+      for role in var.custom_roles : {
+        group = group.id
+        role  = role
+      } if contains(keys(local.custom_role_assignments), k)
+  ]]))
+}
+
+
+resource "azurerm_role_assignment" "custom_role_assignments_readers" {
+  for_each = { for k, v in local.role_assignments : "${k}-${v.group}" => v }
+
+
+  principal_id       = azuread_group.readers[each.value.group].object_id
```

Comment:
bit confused with what you've done here, it's access a bunch of readers groups?

Reply:
this is assigning custom roles to the Management group's readers group